packages/app/features/dashboard/beat/index.tsx

```diff
@@ -28,52 +29,54 @@ export default function BeatScreen(props: Props) {
       return (
         <>
           <View className="flex-row w-full">
             <View className="flex flex-1 mr-2 items-center justify-between">
-              <Image
-                uri={imageUrlMedium(imageUrl)}
-                fallbackUri={imageSrc(imageUrl)}
-                className="aspect-square max-w-125 max-h-125 w-full"
-              />
-              <Details code={code} issuer={issuer} />
+              {!entry && <BeatSkeleton style="image" />}
+              {entry && (
+                <Image
+                  uri={imageUrlMedium(entry.imageUrl)}
+                  fallbackUri={imageSrc(entry.imageUrl)}
+                  className="aspect-square max-w-125 max-h-125 w-full"
+                />
+              )}
+              {!entry && (
+                <BeatSkeleton style="collapsable" secondContainer="w-145" />
+              )}
+              {entry && <Details code={entry.code} issuer={entry.issuer} />}
             </View>
             <BeatSummaryColumn entry={entry} holders={details?.holders} />
           </View>
-          {/* TODO skeleton */}
-          {!details && <ActivityIndicator className="mt-5 mx-auto" />}
+          {!details && <BeatSkeleton style="collapsable" />}
           {details?.offers && <Offers offers={details.offers} />}
+          {!details && <BeatSkeleton style="collapsable" />}
           {details?.history && <Activity activities={details.history} />}
         </>
       );
     } else {
       return (
         <>
-          <Image
-            uri={imageUrlMedium(imageUrl)}
-            fallbackUri={imageSrc(imageUrl)}
-            className="aspect-square max-w-125 max-h-125 w-full mb-3"
-          />
-          <BeatSummaryColumn entry={entry} holders={details?.holders} />
-          <Details code={code} issuer={issuer} />
-          {/* TODO skeleton */}
-          {!details && <ActivityIndicator className="mt-5 mx-auto" />}
-          {details?.offers && <Offers offers={details.offers} />}
-          {details?.history && <Activity activities={details.history} />}
+          <View className="flex flex-col w-full">
+            {!entry && <BeatSkeleton style="imageMobile" />}
+            {entry && (
+              <Image
+                uri={imageUrlMedium(entry.imageUrl)}
+                fallbackUri={imageSrc(entry.imageUrl)}
+                className="aspect-square max-w-125 max-h-125 w-full mb-3"
+              />
+            )}
+            <BeatSummaryColumn entry={entry} holders={details?.holders} />
+            {!entry && <BeatSkeleton style="collapsableMobile" />}
+            {entry && <Details code={entry.code} issuer={entry.issuer} />}
+            {!details && <BeatSkeleton style="collapsableMobile" />}
+            {details?.offers && <Offers offers={details.offers} />}
+            {!details && <BeatSkeleton style="collapsableMobile" />}
+            {details?.history && <Activity activities={details.history} />}
+          </View>
```

Comment:
You are using this component multiple times, which results in creating multiple shared values so the animation is not synchronized. My proposal would be:
1. Create one skeleton component for the entire page when the entry is being loaded
2. Create a skeleton for columns it takes longer to load (activities, owners, etc.). You can rearrange column's layout if it would make the task easier